lib/smooth/core/io/i2c/I2CMasterDevice.cpp

```diff
@@ -126,26 +197,36 @@ namespace smooth::core::io::i2c
 
         // Complete the read with a stop condition.
         res |= i2c_master_stop(link);
+
+        // Send the queued commands
         res |= i2c_master_cmd_begin(port, link, to_tick(timeout));
 
+        // If we implemented slave clock-stretching then restore master timeout to its original value
+        if (scl_timeout > 0)
+        {
+            res |= i2c_set_timeout(port, orig_scl_timeout);
+        }
+
         if (res != ESP_OK)
         {
-            std::stringstream ss;
-            ss << "Error during read of address 0x" << std::hex << static_cast<int32_t>(address);
-            log_error(res, ss.str().c_str());
-            i2c_reset_tx_fifo(port);
-            i2c_reset_rx_fifo(port);
+            log_error(res, address);
+            do_cleanup();
         }
 
         return res == ESP_OK;
     }
 
+    // Write read address and read data
     bool I2CMasterDevice::read_block(uint8_t address,
-                                     core::util::FixedBufferBase<uint8_t>& data,
+                                     FixedBufferBase<uint8_t>& data,
```

Comment:
Same here, && instead of |=. Sorry you'll have to clean up my mistake in this PR :)